src/openrct2/world/Banner.cpp

```diff
@@ -128,6 +128,27 @@ TileElement* banner_get_tile_element(BannerIndex bannerIndex)
     return nullptr;
 }
 
+TileElement* banner_get_scrolling_wall_tile_element(BannerIndex bannerIndex)
```

Comment:
Not sure if adding this helper method was the right way to do it, but I didn't want to duplicate the logic for finding the correct wall piece in both the query & execute of the action.

Reply:
This is good, it should return a WallTileElement as we know that is all we get.